starter-code/src/App.js

```diff
@@ -1,18 +1,91 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from "react";
+import FoodBox from "./components/FoodBox";
+import AddFood from "./components/AddFood";
+import SearchBar from "./components/SearchBar";
+
+import foods from "./foods.json";
+
+import "./App.css";
+import "bulma/css/bulma.css";
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      foods: foods,
+      showedFoodform: false,
+      searchquery: ""
+    };
+  }
+
+  showFoodform = () => {
+    this.setState({ showedFoodform: true });
+  };
+
+  handleClick = () => {
+    const myFoodsCopy = { ...foods };
+    myFoodsCopy.quantity += 1;
+    this.setState({ foods: myFoodsCopy });
+  };
+
+  addFoodHandler = theFood => {
+    const myFoodsCopy = [...foods];
+    myFoodsCopy.push(theFood);
+    this.setState({ foods: myFoodsCopy, showedFoodform: false });
+  };
+
+  updateSearchQuery = newquery => {
+    this.setState({
+      searchquery: newquery
+    });
+  };
+
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+      <div className="container">
+        <h1 className="title">IRON NUTRITION</h1>
+        <div>
+          <SearchBar update={this.updateSearchQuery} />
+        </div>
+
+        <div className="columns">
+          <div className="column">
+            {this.state.foods
+              .filter(food =>
+                food.name
+                  .toLowerCase()
+                  .includes(this.state.searchquery.toLowerCase())
```

Comment:
En faisant un `filter` dans le render avec quelque chose qui dépend du state, tu vas avoir des problèmes car react ne va pas bien rafraichir ta liste.
Ce que tu peux faire à la place c'est de faire du conditional rendering ligne 60: `return food.name.toLowerCase().includes(this.state.searchquery.toLowerCase()) ? <FoodBox key={index} food={food} /> : ''`